interfaces/python/CMakeLists.txt

```diff
@@ -7,284 +7,272 @@
-            )
-
-            configure_file(${CMAKE_CURRENT_SOURCE_DIR}/setup.py.in
-                           ${CMAKE_CURRENT_BINARY_DIR}/setup.py)
-            configure_file(${CMAKE_CURRENT_SOURCE_DIR}/helics.py
-                           ${CMAKE_CURRENT_BINARY_DIR}/helics.py COPYONLY)
+    # Setup rpath
+    set(CMAKE_SKIP_BUILD_RPATH FALSE)
+    set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+    set(CMAKE_MACOSX_RPATH ON)
+    set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
-        endif()
+    # Add the local directory to the rpath
+    if(NOT APPLE)
+        list(APPEND CMAKE_INSTALL_RPATH $ORIGIN)
+        list(APPEND CMAKE_INSTALL_RPATH
+                    "\$ORIGIN/../${CMAKE_INSTALL_LIBDIR}"
+        )
     else()
+        list(APPEND CMAKE_INSTALL_RPATH "@loader_path/.")
+        list(APPEND CMAKE_INSTALL_RPATH "@executable_path/.")
+    endif()
+endif()
+
+###############################
+# Find required dependencies
+###############################
+# Find HELICS
+if(NOT hasParent OR NOT TARGET HELICS::helicsSharedLib)
+    find_package(HELICS 2 REQUIRED)
+endif()
+
+if(HELICS_ENABLE_SWIG OR HELICS_SWIG_GENERATE_INTERFACE_FILES_ONLY)
+    # Find SWIG
+    if(NOT DEFINED SWIG_FOUND OR SWIG_VERSION VERSION_LESS 3)
+        if(POLICY CMP0078)
+            cmake_policy(SET CMP0078 OLD)
+        endif()
+        if(POLICY CMP0086)
+            cmake_policy(SET CMP0086 OLD)
+        endif()
+        find_package(SWIG 3 REQUIRED)
+        include(${SWIG_USE_FILE})
+    endif()
 
-        set(
-            Python_ADDITIONAL_VERSIONS
-            3.4
-            3.5
-            3.6
-            3.7
-            3.8
-        )
-        if(NOT PYTHON_EXECUTABLE)
-            find_package(PythonInterp 3 REQUIRED)
-            if(PYTHONINTERP_FOUND)
-                set(Python_ADDITIONAL_VERSIONS "${PYTHON_VERSION_STRING}")
+    # Find helics.i swig interface file
+    if(NOT HELICS_SWIG_helics.i_FILE)
+        foreach(TMPDIR IN LISTS HELICS_SWIG_INCLUDE_DIRS)
+            if(NOT HELICS_SWIG_helics.i_FILE)
+                if(EXISTS "${TMPDIR}/helics.i")
+                    # cmake-format: off
+                    set(HELICS_SWIG_helics.i_FILE "${TMPDIR}/helics.i" CACHE FILEPATH "Path to the helics.i file used by swig" FORCE)
+                    # cmake-format: on
+                endif()
             endif()
+        endforeach()
+        if(NOT HELICS_SWIG_helics.i_FILE)
+            message(FATAL_ERROR "Unable to find helics.i file, required for swig to generate interfaces")
         endif()
-        find_package(PythonLibs 3 REQUIRED)
+    endif()
+endif()
+
+# CMake UseSWIG doesn't properly propagate include dirs when using target_link_libraries
+# This is also the only way that works with CMake < 3.12
+include_directories(${HELICS_SWIG_INCLUDE_DIRS})
+
+###############################
+# UseSWIG feature support
+###############################
+# This can be removed in HELICS 3 if the minimum CMake version is >= 3.12
```

Comment:
I think at least for the first while the minimum cmake will move to 3.10 for HELICS 3.  I want to try to keep it able to build natively and easily on Ubuntu 18.04.